stdweb-derive/src/lib.rs

```diff
@@ -69,6 +70,18 @@ pub fn derive_reference_type( input: TokenStream ) -> TokenStream {
                         panic!( "The value of '#[reference(instance_of = ...)]' is not a string!" );
                     }
                 },
+                syn::NestedMeta::Meta( syn::Meta::NameValue( ref meta ) ) if meta.ident == "constraint" => {
+                    if let syn::Lit::Str( ref str ) = meta.lit {
+                        let val = str.value();
+                        let parts : Vec<&str> = val.splitn(2,"=").collect();
+                        if parts.len() != 2 {
```

Comment:
AFAIR won't `splitn(2, ...)` always return two elements? If so then this will never trigger.